src/main/java/com/mercadolibre/grupo1/projetointegrador/services/ProductService.java

```diff
@@ -0,0 +1,56 @@
+package com.mercadolibre.grupo1.projetointegrador.services;
+
+import com.mercadolibre.grupo1.projetointegrador.dtos.ProductDTO;
+import com.mercadolibre.grupo1.projetointegrador.entities.Product;
+import com.mercadolibre.grupo1.projetointegrador.entities.enums.ProductCategory;
+import com.mercadolibre.grupo1.projetointegrador.exceptions.ExceptionCatchIsEmpty;
+import com.mercadolibre.grupo1.projetointegrador.exceptions.ExceptionCatchStatusCategory;
+import com.mercadolibre.grupo1.projetointegrador.repositories.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+/*
+@author Gabriel Essenio
+Construindo service de para tratar as requisitos da busca de produtos
+ */
+@Service
+public class ProductService {
+    /*
+    Faz injeção de dependecia do repositorio de produtos
+     */
+    @Autowired
+    private ProductRepository productRepository;
+    /*
+    Metodo que chama Repositorio de Produto e retorna todos produtos cadastrados
+     */
+    public List<ProductDTO> listAllProducts() throws ExceptionCatchIsEmpty {
```

Comment:
tira esse throws como é uma RuntimeException ela vai ser capturada no Advice.
```suggestion
    public List<ProductDTO> listAllProducts(){
```

Reply:
👍